rosidl_generator_java/resource/action.cpp.em

```diff
@@ -1,23 +1,108 @@
 @# Included from rosidl_generator_java/resource/idl.cpp.em
 @{
+import os
+
+from rosidl_cmake import expand_template
 from rosidl_generator_c import idl_structure_type_to_c_include_prefix
 
-action_includes = [
+namespaces = action.namespaced_type.namespaces
+type_name = action.namespaced_type.name
+goal_type_name = action.goal.structure.namespaced_type.name
+result_type_name = action.result.structure.namespaced_type.name
+feedback_type_name = action.feedback.structure.namespaced_type.name
+feedback_message_type_name = action.feedback_message.structure.namespaced_type.name
+send_goal_type_name = action.send_goal_service.namespaced_type.name
+get_result_type_name = action.get_result_service.namespaced_type.name
+
+data = {
+    'package_name': package_name,
+    'output_dir': output_dir,
+    'template_basepath': template_basepath,
+}
+
+# Generate Goal message type
+data.update({'message': action.goal})
+output_file = os.path.join(
+    output_dir, *namespaces[1:], '{0}.ep.{1}.cpp'.format(goal_type_name, typesupport_impl))
```

Comment:
Just a question; why the ".ep." portion of the filename?

Reply:
`ep` originally stood for entry point, so that code could be split between common code and for each typesupport implementation

Reply:
If there are suggestions for renaming (or removing it) I'm happy to do so.